lua/lspconfig/server_configurations/delphi_ls.lua

```diff
@@ -0,0 +1,25 @@
+local util = require("lspconfig.util")
+
+return {
+  default_config = {
+    cmd = { "DelphiLSP.exe" },
+    filetypes = { "pascal" },
+    root_dir = util.root_pattern("*.dpr"),
+    single_file_support = true,
+    settings = {
+      --look for *.delphilsp.json - Delphi lsp config file
+      settingsFile = vim.fs.find(function(name)
+        return name:match(".*%.delphilsp.json$")
+      end, { limit = math.huge, type = "file", upward = true })[1],

```

Comment:
it's better add stop condition

Reply:
Yes, I agree, the limit was is slightly too big 😅 . Set it to 3.